debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java

```diff
@@ -0,0 +1,416 @@
+        res.close();
+        return firstScnOfOnlineLog;
+    }
+
+    /**
+     * Sets NLS parameters for mining session.
+     *
+     * @param connection session level database connection
+     * @throws SQLException if anything unexpected happens
+     */
+    static void setNlsSessionParameters(JdbcConnection connection) throws SQLException {
+        connection.executeWithoutCommitting(SqlUtils.NLS_SESSION_PARAMETERS);
+    }
+
+    /**
+     * This is to update MBean metrics associated with REDO LOG groups
+     * @param connection connection
+     * @param fileNames name of current REDO LOG files
+     * @param metrics current metrics
+     */
+    private static void updateRedoLogMetrics(Connection connection, LogMinerMetrics metrics, Set<String> fileNames) {
+        try {
+            // update metrics
+            Map<String, String> logStatuses = getRedoLogStatus(connection);
+            metrics.setRedoLogStatus(logStatuses);
+
+            int counter = getSwitchCount(connection);
+            metrics.setSwitchCount(counter);
+            metrics.setCurrentLogFileName(fileNames);
+        } catch (SQLException e) {
+            LOGGER.error("Cannot update metrics");
+        }
+    }
+
+    /**
+     * This fetches online redo log statuses
+     * @param connection privileged connection
+     * @return REDO LOG statuses Map, where key is the REDO name and value is the status
+     * @throws SQLException if anything unexpected happens
+     */
+    private static Map<String, String> getRedoLogStatus(Connection connection) throws SQLException {
+        return getMap(connection, SqlUtils.REDO_LOGS_STATUS, UNKNOWN);
+    }
+
+    /**
+     * This fetches REDO LOG switch count for the last day
+     * @param connection privileged connection
+     * @return counter
+     */
+    private static int getSwitchCount(Connection connection) {
+        try {
+            Map<String, String> total = getMap(connection, SqlUtils.SWITCH_HISTORY_TOTAL_COUNT, UNKNOWN);
+            if (total != null && total.get("total") != null){
+                return Integer.parseInt(total.get("total"));
+            }
+        } catch (Exception e) {
+            LOGGER.error("Cannot get switch counter due to the {}", e);
+        }
+        return 0;
+    }
+
+    /**
+     * This method checks if supplemental logging was set on the database level. This is critical check, cannot work if not.
+     * @param connection oracle connection on logminer level
+     * @param pdbName pdb name
+     * @throws SQLException if anything unexpected happens
+     */
+    static void checkSupplementalLogging(OracleConnection connection, String pdbName) throws SQLException {
+        try {
+            if (pdbName != null) {
+                connection.setSessionToPdb(pdbName);
+            }
+
+            final String key = "KEY";
+            String validateGlobalLogging = "SELECT '" + key + "', " + " SUPPLEMENTAL_LOG_DATA_ALL from V$DATABASE";
```

Comment:
cleanup

Reply:
I don't see a problem with this as the variable is used later in an equality check.

Reply:
I mean all the string concattenation on one line. No biggy but I just noticed it.